src/NLog/Internal/MultiFileWatcher.cs

```diff
@@ -46,12 +46,22 @@ namespace NLog.Internal
     /// </summary>
     internal class MultiFileWatcher : IDisposable
     {
-        private List<FileSystemWatcher> watchers = new List<FileSystemWatcher>();
+        private Dictionary<string, FileSystemWatcher> watcherMap = new Dictionary<string, FileSystemWatcher>();
+
+        /// <summary>
+        /// The types of changes to watch for.
+        /// </summary>
+        public NotifyFilters NotifyFilters { get; set; }
 
         /// <summary>
         /// Occurs when a change is detected in one of the monitored files.
         /// </summary>
-        public event EventHandler OnChange;
+        public event FileSystemEventHandler OnChange;
+        
+        public MultiFileWatcher(NotifyFilters notifyFilters = NotifyFilters.LastWrite | NotifyFilters.CreationTime | NotifyFilters.Size | NotifyFilters.Security | NotifyFilters.Attributes)
```

Comment:
note this is **not** binary compatible, because optional arguments are filled at compile phase. Please remove the optional, and add an default ctor for backwardscompatiblity.